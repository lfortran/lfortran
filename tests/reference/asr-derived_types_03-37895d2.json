{
    "basename": "asr-derived_types_03-37895d2",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/derived_types_03.f90",
    "infile_hash": "fd0a87db249f310cd20a90b39d5eb4627597acf774b62bdb8c4256c4",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-derived_types_03-37895d2.stdout",
    "stdout_hash": "e3efc3243d0e6216648c575b09188f6fc9c3bf9cd1541ed6c7ad62f2",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}