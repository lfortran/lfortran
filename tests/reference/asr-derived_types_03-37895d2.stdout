(TranslationUnit
    (SymbolTable
        1
        {
            derived_types_03:
                (Program
                    (SymbolTable
                        2
                        {
                            b:
                                (Variable
                                    2
                                    b
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Struct
                                        2 x
                                        []
                                    )
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            x:
                                (StructType
                                    (SymbolTable
                                        3
                                        {
                                            i:
                                                (Variable
                                                    3
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    x
                                    []
                                    [i]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    ()
                                    ()
                                ),
                            y:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            a:
                                                (StructType
                                                    (SymbolTable
                                                        5
                                                        {
                                                            i:
                                                                (Variable
                                                                    5
                                                                    i
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4 [])
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                )
                                                        })
                                                    a
                                                    []
                                                    [i]
                                                    Source
                                                    Public
                                                    .false.
                                                    .false.
                                                    ()
                                                    ()
                                                ),
                                            b:
                                                (Variable
                                                    4
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        4 a
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    y
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                ),
                            z:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            a:
                                                (StructType
                                                    (SymbolTable
                                                        7
                                                        {
                                                            i:
                                                                (Variable
                                                                    7
                                                                    i
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4 [])
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                )
                                                        })
                                                    a
                                                    []
                                                    [i]
                                                    Source
                                                    Public
                                                    .false.
                                                    .false.
                                                    ()
                                                    ()
                                                ),
                                            b:
                                                (Variable
                                                    6
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        6 a
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            z:
                                                (Variable
                                                    6
                                                    z
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    z
                                    (FunctionType
                                        []
                                        (Integer 4 [])
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 6 z)
                                        (IntegerConstant 5 (Integer 4 []))
                                        ()
                                    )]
                                    (Var 6 z)
                                    Public
                                    .false.
                                    .false.
                                )
                        })
                    derived_types_03
                    []
                    []
                )
        })
    []
)
